### ISSUE TITLE
unbound	blocklist download failures: error : HTTPSConnectionPool: Read timed out.

### ISSUE THREAD

**Issue by CallMeR** [opnsense/core#7371]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [*] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [*] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

1. Unbound often logs download failures, but the download can be completed using a Windows browser.  

2. Unbound will use the content of incompletely downloaded files and import them into the DNSBL list.  

Perhaps these list files are relatively large, or the file links are behind a CDN network. 

However, Unbound's timeout retry strategy is somewhat aggressive, with a timeout of only 5 seconds, and it will use incompletely downloaded files, resulting in an incomplete list

*Tip*: to validate your setup was working with the previous version, use opnsense-revert (https://docs.opnsense.org/manual/opnsense_tools.html#opnsense-revert)

**To Reproduce**

1. Add the following custom list links in the Unbound DNSBL  
```bash
https://malware-filter.gitlab.io/malware-filter/urlhaus-filter-hosts.txt
https://anti-ad.net/domains.txt
https://neodev.team/lite_host
```

2. Click the "Apply" button to let Unbound start downloading the target file  

![20240410192457](https://github.com/opnsense/core/assets/11908641/a4ee4432-0b29-42bf-b9c8-b13a238b9db4)

**Expected behavior**

1. Follow the CDN associated with the list download link.  

2. Wait for the file download to complete.  

3. Adjust the timeout and retry strategy.  

4. If unable to download the complete list in the end, consider abandoning the current list update instead of importing an incomplete list.

**Environment**

Software version used and hardware type if relevant, e.g.:

OPNsense 24.1.5_3-amd64  
FreeBSD 13.2-RELEASE-p11  
OpenSSL 3.0.13

**Comment by AdSchellevis** [comment]
changing the timeout is relatively easy:

https://github.com/opnsense/core/blob/1afe0405bcebd7453790d1ef61b6d262a4d78072/src/opnsense/scripts/unbound/blocklists/__init__.py#L111

I don't think we should try to modify the streaming behavior of the script, when for example 80% of the file can be downloaded, it's usually better to process it than ignore it in full. These sets don't really represent a consistent state that is only valid when processed in full.....

**Comment by CallMeR** [comment]
> it's usually better to process it than ignore it in full

It does make sense in most scenarios, but if 90% was downloaded the first time and only 10% this time, it would result in the 10% of valid data overwriting the initial 90% of valid data.

**Comment by AdSchellevis** [comment]
True, I missed the caching part in the code here 

https://github.com/opnsense/core/blob/b551927c895de141dde21f5dca54a4503ad60e01/src/opnsense/scripts/unbound/blocklists/default_bl.py#L124-L130

**Comment by CallMeR** [comment]
I just went to check my DNSBL list again, and it seems that this time I have encountered a more serious download problem. 

My list is now left with only 1000+ entries.

Currently, I am updating the DNSBL every 8 hours through a scheduled task, but it doesn't seem to have solved this problem. 

According to the code you provided earlier, the cache TTL is set to 20 hours. 

This means that I may have inadvertently missed almost a day's worth of data, causing some leaks as that data was not properly blocked.

![image](https://github.com/opnsense/core/assets/11908641/b03ed35f-2d16-47eb-94dd-7b195c855efd)

![image](https://github.com/opnsense/core/assets/11908641/ec3ed84d-dbe5-479c-b671-2e218e0e23b0)
